src/d3a/models/market/market_structures.py

```diff
@@ -296,13 +296,13 @@ def to_json_string(self):
     def serializable_dict(self):
         return {
             "type": "Trade",
-            "match_type": "Offer" if isinstance(self.offer, Offer) else "Bid",
+            "match_type": "Offer" if isinstance(self.offer_bid, Offer) else "Bid",
```

Comment:
I think you can use the classname for this, and thus removing the need for the isinstance.